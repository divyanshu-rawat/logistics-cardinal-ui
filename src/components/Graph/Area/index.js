import React from 'react';
import PropTypes from 'prop-types';
import { Area as RCArea } from 'recharts';

import { getStrokeWidth } from '../utils';
import { SIZES } from '../../utils/constants';

class Area extends RCArea {
  static propTypes = {
    ...RCArea.propTypes,
    /**
     * Stroke color, which would be a name of a color or an rgb()
     */
    color: PropTypes.string,
    /**
     * Filled color, which would be a name of a color or an rgb()
     */
    bgColor: PropTypes.string,
    /**
     * Size of the line
     */
    size: PropTypes.oneOf(Object.values(SIZES)),
  };

  static defaultProps = {
    ...RCArea.defaultProps,
  };

  render() {
    const { color, bgColor, size, ...props } = this.props;
    const additionalProps = {};

    if (color) {
      additionalProps.stroke = color;
    }

    if (bgColor) {
      additionalProps.fill = bgColor;
    }

    if (size) {
      additionalProps.strokeWidth = getStrokeWidth(size);
    }

    return <RCArea {...props} {...additionalProps} />;
  }
}

export default Area;
