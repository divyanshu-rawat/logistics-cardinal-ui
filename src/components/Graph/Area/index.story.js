import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { color, select } from '@storybook/addon-knobs';

import withTests from '../../../utils/withTests';
import { GRAPH } from '../../../../.storybook/sections';

import ComposedChart from '../ComposedChart';
import Area from '.';
import { generateRandomShifts } from '../utils';
import { SIZES } from '../../utils/constants';

storiesOf(`${GRAPH}Area`, module)
  .addDecorator(withTests('Area/tests/index'))
  .add(
    'Simple',
    withInfo()(() => (
      <ComposedChart data={generateRandomShifts()}>
        <Area
          dataKey="shifts"
          color={color('color', '#78EDA4')}
          bgColor={color('bgColor', '#76E6AC')}
          size={select('size', Object.values(SIZES), SIZES.xs)}
        />
      </ComposedChart>
    )),
  );
