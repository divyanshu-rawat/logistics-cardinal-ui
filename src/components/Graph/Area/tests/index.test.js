import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Area from '..';

const renderedComponent = () =>
  shallow(
    <Area dataKey="shifts" color="chocolate" bgColor="papayawhip" size="xs" />,
  );

describe('<Area />', () => {
  it('should render the component', () => {
    const wrapper = renderedComponent();
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
});
