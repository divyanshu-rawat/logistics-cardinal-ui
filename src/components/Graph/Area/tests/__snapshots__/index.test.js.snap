// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Area /> should render the component 1`] = `
<Area
  activeDot={true}
  animationBegin={0}
  animationDuration={1500}
  animationEasing="ease"
  connectNulls={false}
  dataKey="shifts"
  dot={false}
  fill="papayawhip"
  fillOpacity={0.6}
  hide={false}
  isAnimationActive={true}
  legendType="line"
  points={Array []}
  stroke="chocolate"
  strokeWidth={1}
  xAxisId={0}
  yAxisId={0}
/>
`;
